packages/playwright-core/src/client/elementHandle.ts

```diff
@@ -262,16 +262,42 @@ export async function convertInputFiles(files: string | FilePayload | string[] |
   if (items.some(item => typeof item === 'string')) {
     if (!items.every(item => typeof item === 'string'))
       throw new Error('File paths cannot be mixed with buffers');
+    const directoryCount = (await Promise.all(items.map(async item => (await fs.promises.stat(item as string)).isDirectory()))).filter(Boolean).length;
+    if (directoryCount > 1)
+      throw new Error('Only one directory can be uploaded at a time');
 
     if (context._connection.isRemote()) {
-      const streams: channels.WritableStreamChannel[] = await Promise.all((items as string[]).map(async item => {
-        const lastModifiedMs = (await fs.promises.stat(item)).mtimeMs;
-        const { writableStream: stream } = await context._wrapApiCall(() => context._channel.createTempFile({ name: path.basename(item), lastModifiedMs }), true);
-        const writable = WritableStream.from(stream);
-        await pipelineAsync(fs.createReadStream(item), writable.stream());
-        return stream;
-      }));
-      return { streams };
+      let streams: channels.WritableStreamChannel[] | undefined;
+      let localPaths: string[] | undefined;
+      for (const item of items as string[]) {
```

Comment:
I think we should keep `Promise.all()` and extract a method here. It is getting complicated.

Reply:
I simplified it a little bit but now there are a bunch of `if isDirectory` checks. I guess no way around a method. What about moving the map callback into an own method accepting the item, streams and localPath?

Reply:
Reverted to how it was before.